src/components/activity-page/activity-page-content.tsx

```diff
@@ -39,41 +39,36 @@ export class ActivityPageContent extends React.PureComponent <IProps, IState> {
     const primaryEmbeddables = page.embeddables.filter((e: any) => e.section === EmbeddableSections.Interactive);
     const secondaryEmbeddables = page.embeddables.filter((e: any) => (e.section !== EmbeddableSections.Interactive && e.section !== EmbeddableSections.Introduction));
 
-    const pinOffSet = page.layout !== PageLayouts.FullWidth && secondaryEmbeddables.length ? scrollOffset : 0;
-    const pageSectionQuestionCount = getPageSectionQuestionCount(page);
     const questionsBeforePrimary = totalPreviousQuestions + pageSectionQuestionCount.Header
                                    + (primaryFirst ? 0 : pageSectionQuestionCount.InfoAssessment);
-    const renderPrimary = this.renderPrimaryEmbeddables(primaryEmbeddables, questionsBeforePrimary, vertical, primaryFirst && !vertical, pinOffSet);
+    const primaryIsOnLeft = page.layout === PageLayouts.FortySixty;
+    const pinOffSet = page.layout !== PageLayouts.FullWidth && secondaryEmbeddables.length ? scrollOffset : 0;
+    const renderPrimary = this.renderPrimaryEmbeddables(primaryEmbeddables, questionsBeforePrimary, page.layout, primaryIsOnLeft, pinOffSet);
 
-    const collapsible = page.toggle_info_assessment && page.layout !== PageLayouts.FullWidth;
     const questionsBeforeSecondary = totalPreviousQuestions + pageSectionQuestionCount.Header
                                      + (primaryFirst ? pageSectionQuestionCount.InteractiveBlock : 0);
-    const renderSecondary = this.renderSecondaryEmbeddables(secondaryEmbeddables, questionsBeforeSecondary, !primaryFirst && !vertical, collapsible);
+    const secondaryIsOnLeft = page.layout === PageLayouts.Responsive || page.layout === PageLayouts.SixtyForty;
+    const collapsible = page.toggle_info_assessment && page.layout !== PageLayouts.FullWidth;
+    const renderSecondary = this.renderSecondaryEmbeddables(secondaryEmbeddables, questionsBeforeSecondary, page.layout, secondaryIsOnLeft, collapsible);
 
     const [first, second] = primaryFirst
                             ? [renderPrimary, renderSecondary]
                             : [renderSecondary, renderPrimary];
 
     return (
-      <div className={`page-content ${useFullPageWidth ? "full" : ""}`} data-cy="page-content">
+      <div className={`page-content ${page.layout === PageLayouts.Responsive ? "full" : ""}`} data-cy="page-content">
         <div className="name">{page.name}</div>
         <div className="introduction">
           { page.text && renderHTML(page.text) }
-          { introEmbeddables && this.renderEmbeddables(introEmbeddables, EmbeddableSections.Introduction, totalPreviousQuestions) }
+          { introEmbeddables && this.renderIntroEmbeddables(introEmbeddables, totalPreviousQuestions) }
         </div>
-        <div className={`embeddables ${vertical ? "vertical" : ""}`}>
+        <div className={`embeddables ${page.layout === PageLayouts.FullWidth ? "vertical" : ""}`}>
```

Comment:
All of the changes above (and many below) are simply a refactor to make the layout specs in this module easier to read.